File: boj7/1735.cpp
// 분수 합
// https://www.acmicpc.net/problem/1735

#include <stdio.h>

int Gcd(int x, int y) {
  if (y == 0) return x;
  return Gcd(y, x % y);
}

int A, B, C, D;

int main() {
  scanf("%d%d%d%d", &A, &B, &C, &D);
  int numerator = A * D + B * C;
  int denominator = B * D;

  int gcd = Gcd(numerator, denominator);
  printf("%d %d\n", numerator / gcd, denominator / gcd);
  return 0;
}

File: past_archive/23boj_py/1735.py
# 분수 합
# https://www.acmicpc.net/problem/1735

import sys


def GCD(a, b):
    if b == 0:
        return a
    return GCD(b, a % b)


a, b = map(int, sys.stdin.readline().split())
c, d = map(int, sys.stdin.readline().split())

num = a * d + c * b
den = b * d

k = GCD(num, den)

sys.stdout.write(f'{num // k} {den // k}\n')
